builtin/logical/aws/client.go

```diff
@@ -23,91 +24,139 @@ import (
 	"github.com/hashicorp/vault/sdk/logical"
 )
 
+// Return a slice of *aws.Config, based on descending configuration priority. STS endpoints are the only place this is used.
 // NOTE: The caller is required to ensure that b.clientMutex is at least read locked
-func (b *backend) getRootConfig(ctx context.Context, s logical.Storage, clientType string, logger hclog.Logger) (*aws.Config, error) {
-	credsConfig := &awsutil.CredentialsConfig{}
-	var endpoint string
-	var maxRetries int = aws.UseServiceDefaultRetries
+func (b *backend) getRootConfig(ctx context.Context, s logical.Storage, clientType string, logger hclog.Logger) ([]*aws.Config, error) {
```

Comment:
suggest changing to plural since we are returning an array
```suggestion
func (b *backend) getRootConfigs(ctx context.Context, s logical.Storage, clientType string, logger hclog.Logger) ([]*aws.Config, error) {
```

Reply:
done!